flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBConnectorDeserializationSchema.java

```diff
@@ -600,6 +601,13 @@ private StringData convertToString(BsonValue docObj) {
             return StringData.fromString(
                     convertInstantToZonedDateTime(instant).format(ISO_OFFSET_DATE_TIME));
         }
+        if (docObj.isArray()) {
+            String doc =
+                    docObj.asArray().stream()
+                            .map(v -> convertToString(v).toString())
+                            .collect(Collectors.joining(", "));
+            return StringData.fromString("[" + doc + "]");
+        }
```

Comment:
```suggestion
        if (docObj.isArray()) {
            // convert bson array to json string
            Writer writer = new StringWriter();
            JsonWriter jsonArrayWriter = new JsonWriter(writer) {
                @Override
                public void writeStartArray() {
                    doWriteStartArray();
                    setState(State.VALUE);
                }
                @Override
                public void writeEndArray() {
                    doWriteEndArray();
                    setState(getNextState());
                }
            };

            new BsonArrayCodec().encode(jsonArrayWriter, docObj.asArray(), EncoderContext.builder().build());
            return StringData.fromString(writer.toString());
        }
```

Reply:
Thanks @vanliu-tx for this contribution.
How about use suggestion's code to keep the BsonArray's format so that we can use `BsonArray.parse(json)` to restore it.